frontend/src/maps/MapView.tsx

```diff
@@ -10,20 +10,62 @@ import UserFocusView from "./widgets/UserFocusView.tsx";
 import PanelButton from "./widgets/PanelButton.tsx";
 import MapPanel from "./MapPanel.tsx";
 import { useRouteOperations } from "./manage/operations.ts";
+import { useEffect } from "react";
 
-const MapView = () => {
+import type { Point, RouteSegment } from "./manage/structure";
+import { useRef } from "react";
+
+const MapView = ({
+  onRouteDataChange,
+  initialData,
+}: {
+  onRouteDataChange: (data: {
```

Comment:
See comment in RoutePlanningPage, it would be preferable to me to fully remove onRouteDataChange